pkg/moov/schedules/schedules_model.go

```diff
@@ -0,0 +1,127 @@
+package schedules
+
+import "time"
+
+type Schedule struct {
+	// prod or sandbox
+	Mode string `json:"mode,omitempty" spanner:"mode"`
+
+	ScheduleID string `json:"scheduleID,omitempty" spanner:"schedule_id"`
+
+	// This is the account ID of the source transfers that the transfer will run using.
+	SourceAccountID string `json:"sourceAccountID,omitempty" spanner:"source_account_id"`
+
+	// This is the destination ID of the destination transfers that the transfer will run using.
+	DestinationAccountID string `json:"destinationAccountID,omitempty" spanner:"destination_account_id"`
+
+	// This is the partner ID that the transfer will run using.
+	PartnerAccountID string `json:"partnerAccountID,omitempty" spanner:"partner_account_id"`
+
+	// AccountID of the account that created it and is allowed to update it.
+	OwnerID string `json:"ownerID,omitempty" spanner:"owner_id"`
+
+	// Description of what this schedule is
+	Description string `json:"description,omitempty" spanner:"description"`
+
+	// If specified will generate Scheduled transfers based on its configuration
+	RecurTransfer *RecurTransfer `json:"recurTransfer,omitempty" spanner:"recur_transfer"`
+
+	// List of all generated and manually added transfers to be made.
+	Occurrences []TransferOccurrence `json:"occurrences,omitempty" spanner:"-"`
+
+	CreatedOn  time.Time  `json:"createdOn,omitempty" spanner:"created_on"`
+	UpdatedOn  time.Time  `json:"updatedOn,omitempty" spanner:"updated_on"`
+	DisabledOn *time.Time `json:"disabledOn,omitempty"  spanner:"disabled_on"`
+}
+
+func (s Schedule) ToUpsertSchedule() UpsertSchedule {
+	occs := make([]UpsertTransferOccurrence, len(s.Occurrences))
+	for i, occ := range s.Occurrences {
+		occs[i] = UpsertTransferOccurrence{
+			OccurrenceID: occ.OccurrenceID,
+			Transfer:     occ.Transfer,
+			RunOn:        occ.RunOn,
+			Cancelled:    nil,
+		}
+	}
+
+	return UpsertSchedule{
+		Description:   s.Description,
+		RecurTransfer: s.RecurTransfer,
+		Occurrences:   occs,
+	}
+}
+
+// https://www.rfc-editor.org/rfc/rfc5545#section-3.3.10
+type RecurTransfer struct {
+	// Transfer values to use to create the transfer based on the recurRule
+	// When changed, should just modify the transfer of the schedules
+	Transfer Transfer `json:"transfer,omitempty"`
+
+	// https://www.rfc-editor.org/rfc/rfc5545#section-3.3.10
+	RecurrenceRule string `json:"recurrenceRule,omitempty"`
```

Comment:
We need some examples in the docs here. Are we going to offer a generator or suggest a third party library to use to help build?

Reply:
Yeah we're going to offer some guidance. Josh also has me wanting to implement an AI endpoint for it lol.